### PR TITLE
refactor: use fn push_origin to simplify code

### PR DESCRIPTION
**Pull request by jackwener**
use fn push_origin to simplify code

### CODE REVIEW

File: src/vm.rs [hunk 1]
```diff
@@ -166,10 +166,24 @@ impl<'a, S: Storage, C: PevmChain> VmDb<'a, S, C> {
         }
     }
 
+    // a function to push read_origins
```
Comment (hai-rise) [inline src/vm.rs:169]: Let's elaborate a bit!
```
// Push a new read origin. Return an error when there's already 
// an origin but doesn't match the new one to force re-execution.
```

**Review by hai-rise — CHANGES_REQUESTED** [review]
Minor comments but overall really like this simplification. Thank you!

**Comment by jackwener** [comment]
@hai-rise Thank you very much for your careful review!

**Review by hai-rise — APPROVED** [review]
🫡

**Pull request merged**